File: library/base/src/main/java/org/cru/godtools/base/util/FileUtils.java
package org.cru.godtools.base.util;

import android.content.Context;

import java.io.File;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.WorkerThread;

public final class FileUtils {
    @WorkerThread
    public static boolean createResourcesDir(@NonNull final Context context) {
        final File dir = getResourcesDir(context);
        return (dir.exists() || dir.mkdirs()) && dir.isDirectory();
    }

    @NonNull
    public static File getResourcesDir(@NonNull final Context context) {
        return new File(context.getFilesDir(), "resources");
    }

    @Nullable
    public static File getFile(@NonNull final Context context, @Nullable final String name) {
        return name != null ? new File(getResourcesDir(context), name) : null;
    }
}

File: library/base/src/main/java/org/cru/godtools/base/util/FileUtils.kt
@file:JvmName("FileUtils")

package org.cru.godtools.base.util

import android.content.Context
import kotlinx.coroutines.runBlocking
import org.cru.godtools.base.fileManager

@Deprecated("use FileManager to access files instead")
fun Context.getGodToolsFile(name: String?) = name?.let { runBlocking { fileManager.getFile(name) } }
